model/insights.go

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package model
+
+import (
+	"net/http"
+	"time"
+)
+
+const (
+	TimeRange1Day  string = "1_day"
+	TimeRange7Day  string = "7_day"
+	TimeRange28Day string = "28_day"
+)
+
+type InsightsOpts struct {
+	TimeRange int64
```

Comment:
Instead of `TimeRange` being `int64` could we make it a `time.Duration` and add a custom method `(i *InsightsOpts),MarshalJSON` that can convert the parameter we receive from the API request (ex `1_day`) to a `time.Duration` value (ex  `time.Duration(time.Hour*24)`)?

Reply:
Sorry, just remembered it's not coming in via JSON but rather via a query param, so the `MarshalJSON` method wouldn't make sense, but same idea.

Reply:
Right now we're converting the string directly to the timestamp that's used in the query (which is of type `int64`). 
https://github.com/mattermost/mattermost-server/blob/8b39b8be808c9e7ed08d7b1f89815031a727741d/model/insights.go#L36-L47

I'm not sure I see the benefit of storing it as a `time.Duration` if we'd have to convert that into the timestamp before passing it to the query. I might be misunderstanding what you mean though.

Just to clarify as well, `TimeRange` here holds the timestamp of the earliest reaction that we would fetch. It might be a misunderstanding of the name, in which case I can rename it to something more clear.

Reply:
Right, now I realize it's not representing a range at all and I agree that another option is to rename the field to something like `StartUnixMilli` (making sure it includes the units).

What do you prefer from the perspective of a client using `(*App).GetTopReactionsForUserSince`, do you think it's better to say get me reactions for the past `time.Duration(time.Hour)` or to pass in the unix timestamp?